nixos/modules/tasks/lvm.nix

```diff
@@ -7,17 +7,18 @@ in {
   options.services.lvm = {
     package = mkOption {
       type = types.package;
-      default = if cfg.dmeventd.enable then pkgs.lvm2_dmeventd else pkgs.lvm2;
```

Comment:
Good question. I don't think so. `lvm2_dmeventd` is only referenced by the `dmraid` package and in this line, AFAICT.

Reply:
Well, I am not sure what dmeventd does, but it seems plausible that when it enabled you need to use a build of LVM that includes it…

Reply:
Isn't it dependent on vdo, so that dmeventd-only case is not handled?

The problem is that I of course don't use dmeventd so I do not know what is a good enough easy check…

Reply:
It shouldn't be, that's what line 44 does. It's in `(mkIf cfg.dmeventd.enable { […] }`. Then line 89 sets the lvm2_vdo package when vdo is enabled. I think (and hope) I have all the priorities in order here, but vdo (999) should override mkDefault, should override mkOptionDefault.

We use dmeventd to automatically (try to) grow thinpools, once they become too full. I guess I could try to put that into a nixos test.

Reply:
Oh right. Well, I guess having a vdo test if vdo includes dmevents could indeed be convincing (then if something breaks dmeventd, maybe the issue will be convertible to a test).

Reply:
You mean [like this](https://github.com/NixOS/nixpkgs/blob/cfbf8d6c96b7aee0cb75d0eca97bae8ae5f693a8/pkgs/os-specific/linux/vdo/default.nix) or something more sophisticated?

Reply:
I think it is a link to the wrong file? (package not test)

Reply:
Right, sorry. https://github.com/NixOS/nixpkgs/blob/cfbf8d6c96b7aee0cb75d0eca97bae8ae5f693a8/nixos/tests/lvm2/vdo.nix

Reply:
Hm, you said that dmeventd gets used when resizing happens? Does it happen here?

Reply:
No. I meant we as in @helsinki-systems use it for that. I can put that into the thin test or vdo test in the next few days/weeks.